src/commands/force/source/retrieve.ts

```diff
@@ -131,4 +136,10 @@ export class Retrieve extends SourceCommand {
 
     return formatter.getJson();
   }
+
+  private wantsToRetrieveCustomFields(): boolean {
+    return this.componentSet.toArray().some((sourceComponent: SourceComponent) => {
```

Comment:
error condition is

some  (type.name === Customfield && fullname === ComponentSet.WILDCARD)
! some ((type.name === CustomObject && fullname === ComponentSet.WILDCARD))